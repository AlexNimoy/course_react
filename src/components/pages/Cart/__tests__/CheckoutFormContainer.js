import React from 'react';
import { render } from '@testing-library/react';
import 'jest-dom/extend-expect';
import CheckoutFormContainer from '../CheckoutForm';

describe('<CheckoutFormContainer>', () => {
  const form = render(<CheckoutFormContainer />);

  it('from include fields', () => {
    expect(form.getByPlaceholderText('ФИО')).toBeInTheDocument();
    expect(form.getByPlaceholderText('Телефон')).toBeInTheDocument();
    expect(form.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(form.getByPlaceholderText('Адрес')).toBeInTheDocument();
  });

  it('include submit button', () => {
    expect(form.getByText('Submit')).toBeInTheDocument();
    expect(form.getByText('Submit')).toHaveAttribute('type', 'submit')
  });

  it('inclue h1 tag', () => {
    expect(form.getByText('Checkout')).toMatchSnapshot();
  });

  it('not have error div', () => {
    expect(form.queryByTestId('error-text')).not.toBeInTheDocument();
  });
});
