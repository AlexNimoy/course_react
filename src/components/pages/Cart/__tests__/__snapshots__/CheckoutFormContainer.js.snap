// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutFormContainer> inclue h1 tag 1`] = `
<h1>
  Checkout
</h1>
`;
